gozstd.go

```diff
@@ -372,6 +395,10 @@ func getStreamDecompressor(dd *DDict) *streamDecompressor {
 	}
 	sd := v.(*streamDecompressor)
 	sd.zr.Reset((*srcReader)(sd), dd)
+	if limit > 0 {
+		result := C.ZSTD_DCtx_setMaxWindowSize_wrapper(C.uintptr_t(uintptr(unsafe.Pointer(sd.zr.ds))), C.size_t(limit))
```

Comment:
This limits the size of a single frame. A stream may contain big number of frames, so the total decompressed size across all the frames may exceed the `limit`. This limit must be enforced at `streamDecompress()` function while reading the decompressed data to the destination buffer.